applications/welcome/models/access_control.py

```diff
@@ -0,0 +1,61 @@
+#########################################################################
+## This file defines access control methods, including verification, setting and deleting.
+## - access_view can view only.
+## - access_setting can change settings, and view variables/output, but cannot change procedure code.
+## - access_manage can do everything the main user can do.
+#########################################################################
+
+
+#########################################################################
+# Verification methods
+#########################################################################
+
+# return if the user has authority to view device
+def access_view(user_id, device_token):
+    return auth.has_permission(name="view", group_id=(device_token + "_" + user_id), user_id=user_id)
+
```

Comment:
As stated above, if you could comment on why you are using as group id the concatenation of device_token and user_id, it would be great.  I am not saying it's wrong; I am just confused.